src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,142 @@
+
+    /**
+     * print out the lists of tasks
+     */
+    public void list() {
+        System.out.println("Here are your task(s):");
+        for (int i = 0; i < listOfInputs.size(); i++) {
+            System.out.println(((i + 1) + ". " + listOfInputs.get(i).message()));
+        }
+    }
+
+
+    public void find(String word) {
+        boolean hasWord = false;
+        for (int i = 0; i < listOfInputs.size(); i++) {
+            if (listOfInputs.get(i).description.contains(word)) {
+                System.out.println(listOfInputs.get(i).message());
+                hasWord = true;
+            }
+            if (i == listOfInputs.size() - 1 && !hasWord) {
+                System.err.println("No task with the word: " + word);
+                break;
+            }
+        }
+    }
+
+    public void mark( int curr_no) throws IOException {
+        Task task_done = listOfInputs.get(curr_no);
+        System.out.println("Good work!! I have marked it done:\n" + task_done.markedDone());
+        storage.updateData(task_done.message(),filePath);
+    }
+
+    /**
+     * unmark the list as undone once it is called
+     * @param curr_no
+     * @throws IOException
+     */
+    public void unMark(int curr_no) throws IOException {
+        Task task_undone = listOfInputs.get(curr_no);
+        System.out.println("Alrightt! I have marked it undone:\n" + task_undone.markedUndone());
+        storage.updateData(task_undone.message(),filePath);
+    }
+
+    /**
+     * to update on list and display the todo message once it is called
+     * @param str
+     * @return
+     * @throws IOException
+     */
+    public String toDo(String str) throws IOException {
+        Todos todo = new Todos(str);
+        listOfInputs.add(todo);
+        storage.updateData(todo.message() , filePath);
+        return "Okayy!! I've added this task:\n " + todo.message() + "\n You have " + listOfInputs.size() + " tasks in the list.";
+
+    }
+
+    /**
+     * to update on list and display the deadline message once it is called
+     * @param str
+     * @return
+     * @throws IOException
+     */
+    public String deadLine(String str) throws IOException {
+        String[] deadL = str.split("/by ", 2);
+        DeadLine deadLineTemp = new DeadLine(deadL[0], deadL[1]);
+        listOfInputs.add(deadLineTemp);
+        storage.updateData(deadLineTemp.message(),filePath);
+        return "Deadline for this task:\n " + deadLineTemp.message() + "\n You have " + listOfInputs.size() + " tasks in the list.";
+
+    }
+
+    /**
+     * to update on the list and display the event message once it is called
+     * @param str
+     * @return
+     * @throws IOException
+     */
+    public String event(String str) throws IOException {
+        String[] event = str.split("/at ", 2);
+        Event eventTemp = new Event(event[0], event[1]);
+        listOfInputs.add(eventTemp);
+        storage.updateData(eventTemp.message(),filePath);
```

Comment:
As stated in the coding standards, commas should be followed by a space. I've noted similar issues (comma preceded by a space etc) in other places as well, but it should be as follows: `f(arg1, arg2)` for any method/constructor `f`, what do you think?